framework/Steps.py

```diff
@@ -385,8 +386,12 @@ def _localInputAndCheckParam(self,paramInput):
     roles      = set(rolesItem)
     if 'Optimizer' in roles:
       self.samplerType = 'Optimizer'
-      if 'Sampler' in roles:
-        self.raiseAnError(IOError, 'Only Sampler or Optimizer is alloweed for the step named '+str(self.name))
+      if 'Sampler' in roles or 'MCMC' in roles:
```

Comment:
Hmm... Another role? what's the advantage? I am fairly against to create another role

Reply:
Why cant' this be a sampler? If there are tools you need from the Optimizer we can think to move them in the Adaptive...